### PR TITLE
Cleanup docstring in `gammapy.maps`: `coord.py`, `geom.py`, `io.py` and `maps.py`

### PR DESCRIPTION
**Pull request by MRegeard**
This PR is related to issue https://github.com/gammapy/gammapy/issues/4811.

It address the part of the issue concerning gammapy.maps. Especially on `coord.py`, `geom.py`, `io.py` and `maps.py`.

### CODE REVIEW

File: gammapy/maps/coord.py [hunk 1]
```diff
@@ -115,16 +115,17 @@ def phi(self):
 
     @property
     def frame(self):
-        """Coordinate system (str)."""
+        """Coordinate system as a str."""
```
Comment (Astro-Kirsty) [inline gammapy/maps/coord.py:118]:
```suggestion
        """Coordinate system as a string."""
```

File: gammapy/maps/coord.py [hunk 2]
```diff
@@ -115,16 +115,17 @@ def phi(self):
...
 
     @property
     def match_by_name(self):
-        """Boolean flag: axis lookup by name (True) or index (False)."""
+        """Boolean flag: axis lookup by name return True or by index return False."""
         return self._match_by_name
 
     @property
     def skycoord(self):
+        """Coordinate object as an `~astropy.coordinates.SkyCoord`."""
```
Comment (Astro-Kirsty) [inline gammapy/maps/coord.py:128]:
```suggestion
        """Coordinate object as a `~astropy.coordinates.SkyCoord`."""
```

File: gammapy/maps/coord.py [hunk 3]
```diff
@@ -194,12 +201,12 @@ def create(cls, data, frame=None, axis_names=None):
         ----------
         data : tuple, dict, `~gammapy.maps.MapCoord` or `~astropy.coordinates.SkyCoord`
             Object containing coordinate arrays.
-        frame : {"icrs", "galactic", None}, optional
+        frame : {"icrs", "galactic", None}
```
Comment (Astro-Kirsty) [inline gammapy/maps/coord.py:204]:
```suggestion
        frame : {"icrs", "galactic", None}, optional
```

File: gammapy/maps/coord.py [hunk 4]
```diff
@@ -293,7 +300,7 @@ def flat(self):
 
     @property
     def broadcasted(self):
-        """Return broadcasted coords"""
+        """Return broadcasted coordinates"""
```
Comment (Astro-Kirsty) [inline gammapy/maps/coord.py:303]:
```suggestion
        """Return broadcasted coordinates."""
```

File: gammapy/maps/geom.py [hunk 5]
```diff
@@ -88,13 +88,13 @@ def data_nbytes(self, dtype="float32"):
 
         Parameters
         ----------
-        dtype : data-type
+        dtype : data-type, optional
```
Comment (Astro-Kirsty) [inline gammapy/maps/geom.py:91]:
```suggestion
        dtype : str, optional
```

File: gammapy/maps/geom.py [hunk 6]
```diff
@@ -255,8 +255,8 @@ def coord_to_idx(self, coords, clip=False):
             coordinate vector for axis i.
         clip : bool
             Choose whether to clip indices to the valid range of the
-            geometry.  If false then indices for coordinates outside
-            the geometry range will be set -1.
+            geometry. If false then indices for coordinates outside
```
Comment (Astro-Kirsty) [inline gammapy/maps/geom.py:258]:
```suggestion
            geometry. If False then indices for coordinates outside
```

File: gammapy/maps/geom.py [hunk 7]
```diff
@@ -288,16 +288,16 @@ def pix_to_coord(self, pix):
...
 
         Parameters
         ----------
         pix : tuple
             Tuple of pixel coordinates.
         clip : bool
             Choose whether to clip indices to the valid range of the
-            geometry.  If false then indices for coordinates outside
-            the geometry range will be set -1.
+            geometry. If false then indices for coordinates outside
```
Comment (Astro-Kirsty) [inline gammapy/maps/geom.py:299]:
```suggestion
            geometry. If False then indices for coordinates outside
```

File: gammapy/maps/geom.py [hunk 8]
```diff
@@ -565,7 +565,7 @@ def replace_axis(self, axis):
 
     @abc.abstractmethod
     def solid_angle(self):
-        """Solid angle (`~astropy.units.Quantity` in ``sr``)."""
+        """Solid angle as an `~astropy.units.Quantity` object (in ``sr``)."""
```
Comment (Astro-Kirsty) [inline gammapy/maps/geom.py:568]:
```suggestion
        """Solid angle as a `~astropy.units.Quantity` (in ``sr``)."""
```

**Review by Astro-Kirsty — COMMENTED** [review]
Thank you @MRegeard 
A few inline comments!

**Review by Astro-Kirsty — DISMISSED** [review]

**Review by cgalelli — DISMISSED** [review]
@MRegeard , look good. A couple inline comments.

**Review by Astro-Kirsty — APPROVED** [review]

**Review by cgalelli — APPROVED** [review]
Thanks @MRegeard !

**Pull request merged**